config_test.go

```diff
@@ -48,16 +48,16 @@ repos:
 	want := &Config{
 		CacheDir: `/tmp`,
 		Port:     9139,
-		Repos: map[string]Repo{
-			"archlinux": Repo{
+		Repos: map[string]*Repo{
+			"archlinux": &Repo{
 				URL: "http://mirrors.kernel.org/archlinux",
 			},
 		},
 		PurgeFilesAfter: 2592000,
 		DownloadTimeout: 200,
 		Prefetch:        &RefreshPeriod{Cron: "0 0 3 * * * *", TTLUnaccessed: 5, TTLUnupdated: 200},
 	}
-	if !cmp.Equal(*got, *want, cmpopts.IgnoreFields(Config{}, "Prefetch")) {
+	if !cmp.Equal(*got, *want, cmpopts.IgnoreFields(Config{}, "Prefetch"), cmpopts.IgnoreFields(Repo{}, "mutex")) {
 		t.Errorf("got %v, want %v", *got, *want)
 	}
```

Comment:
Adding something like

```suggestion
        if *got.Repos != *want.Repos{
	if !cmp.Equal(*got.Repos,*want.Repos){
		t.Errorf("got %v, want %v", *got.Repos, *want.Repos)
	}
	}
```

Reply:
I believe you are. `IgnoreFields` only ignores the fields that are passed as arguments, which are just the mutexes in this case. You can verify that the other fields are checked by changing one of the timestamps in the expected repo in `TestUpdateRepoMirrorlist`.

Reply:
I am dumb, my fault. The check is already here https://github.com/anatol/pacoloco/pull/70/files/4efa1c0b52860cc8c52a931f8daae3a3ee707c9b#diff-4bb821dfaef1884915ee03c5488bded04807301bc3f41fc7001a078c330ae980L63
Sorry for the bother :facepalm: